client/modules/IDE/components/PreviewFrame.jsx

```diff
@@ -86,9 +78,43 @@ class PreviewFrame extends React.Component {
   }
 
   componentWillUnmount() {
+    window.removeEventListener('message', this.handleConsoleEvent);
     ReactDOM.unmountComponentAtNode(this.iframeElement.contentDocument.body);
   }
 
+  handleConsoleEvent(messageEvent) {
+    if (Array.isArray(messageEvent.data)) {
+      messageEvent.data.forEach((message) => {
```

Comment:
i think this loop could be merged with the second one—put everything in an `every` loop by piecing out the body of the `forEach` and putting it in the `every` loop, and `return false` if there's an infinite loop.

Reply:
done :+1: